ArduCopter/events.cpp

```diff
@@ -281,7 +281,14 @@ void Copter::failsafe_terrain_set_status(bool data_ok)
 void Copter::failsafe_terrain_on_event()
 {
     failsafe.terrain = true;
-    gcs().send_text(MAV_SEVERITY_CRITICAL,"Failsafe: Terrain data missing");
+    if (wp_nav->get_terrain_source() == AC_WPNav::TerrainSource::TERRAIN_FROM_TERRAINDATABASE) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL,"Failsafe: Terrain data missing");
```

Comment:
let's use the GCS_SEND_TEXT macro.

optionally we could save some flash by declaring a const str just above and re-using it.

Reply:
I don't think you need to declare it. just using a string constant will be automatically reused by the compiler.

Reply:
Now using the macro